File: ABC/python/058/A.py
import sys
a,b,c = map(int,input().split())
if a < 0 or b < 0 or c < 0 or a > 100 or b > 100 or c > 100:
    sys.exit()

diff_a_b = b - a
diff_b_c = c - b

if diff_a_b == diff_b_c:
    print("YES")
else:
    print("NO")

File: ABC/ruby/058/A.rb
array = []
array = gets().chomp.split(' ')
exit if !( (array[0] =~ /^[0-9]+$/) && (array[1] =~ /^[0-9]+$/) && (array[2] =~ /^[0-9]+$/) )
A = array[0].to_i
B = array[1].to_i
C = array[2].to_i

exit if !( 1 <= A && 1 <= B && 1 <= C && A <= 100 && B <= 100 && C <= 100 )

puts B - A == C - B ? 'YES' : 'NO'